### PR TITLE
Blog App: Add forms

### PR DESCRIPTION
**Pull request by rmiyoyo**
# Work Done

-  Created a method called `current_user` in my ApplicationController to make current user data available to all controllers.
    -  It will return the first user from the database.
    -  Proper authentication will be added in the upcoming projects.
- Created forms to perform the following functions:
    - Creates a Post on behalf of the `current_user` (use the method that I created in my ApplicationController).
    - Create a comment on behalf of the `current_user` (use the method that I created in my ApplicationController).
- Allow Users to add likes to Posts.

### CODE REVIEW

**Comment by rloterh** [comment]
Hi @rmiyoyo 

Stylelint linter is required but missing! Can you include this ASAP!

**Comment by rmiyoyo** [comment]
> Hi @rmiyoyo
> 
> Stylelint linter is required but missing! Can you include this ASAP!

Adding it immediately

**Comment by rmiyoyo** [comment]
> Hi @rmiyoyo
> 
> Stylelint linter is required but missing! Can you include this ASAP!

Hi @rloterh I have added Stylelint linter as you requested. :)

**Comment by rloterh** [comment]
Hi @rmiyoyo 

Stylelint linter checks is not showing yet. Can we fix this ASAP?

image

**Comment by rmiyoyo** [comment]
> Hi @rmiyoyo
> 
> Stylelint linter checks is not showing yet. Can we fix this ASAP?
> 
> image

Hi @rloterh , my apologies, I did not push linters.yml. Fixed!

**Comment by rloterh** [comment]
Looks great! Thanks

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -13,6 +13,11 @@
...
       <p class="mt-5 ml-3" ><%= @post.text %></p>
     </div>
+
+    <div class="mb-5 mt-5 ml-3 flex gap-5">
+      <a class="border-solid border-4 border-black cursor-pointer"><%= render partial: 'shared/add_like' %></a>
+      <a href="/users/<%= @user.id %>/posts/<%= @post.id %>/comments/new" class="border-4 border-solid border-black"> New comment</a>
+    </div>
   </div>
 
   <%= render partial: 'shared/comment', locals: {post: @post} %>
```
Comment (rloterh) [inline app/views/posts/show.html.erb:24]: - [ ] An error is thrown on the _Post#show_  page just after saving a comment for a given post. Let's ensure to fix it.

image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Rite, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🌴 🚦 🚗 🏫 🏡 _🏃

## Highlights 👍  
- Good jobs o far on form creation!
- Post can be added correctly via post forms!
- Good ReadMe

## Required Changes ♻️
_Kindly check the inline comments under the review._


## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
The _Getting Started_ section is occurring twice in the ReadMe, let's endeavor to consolidate them to keep the ReadMe organized!

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rmiyoyo** [comment]
# Work Done
- Fixed the  error is thrown on the Post#show page

File: app/views/posts/show.html.erb (hunk 1 above)
Comment (Munsa1) [reply app/views/posts/show.html.erb:24]: - [ ] As requested by the previous reviewer, kindly fix the error on the Post#show page. 

Videocast
[cinnamon-2023-12-23T230832+0200.webm](https://github.com/rmiyoyo/blog-app/assets/81932478/cc67fcd9-7f52-461b-ba59-8764bdfa8009)

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @rmiyoyo, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
Passing Linters :heavy_check_mark: 
Correct Git Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rmiyoyo** [comment]
# Work Done
- I have fixed the error is thrown on the Post#show page by passing the comment variable to the partial when rendering it.

File: app/views/posts/show.html.erb [hunk 2]
```diff
@@ -6,14 +6,19 @@
...
 
     <div>
       <p class="mt-5 ml-3" ><%= @post.text %></p>
     </div>
+
+    <div class="mb-5 mt-5 ml-3 flex gap-5">
+      <a class="border-solid border-4 border-black cursor-pointer"><%= render partial: 'shared/add_like' %></a>
+      <a href="/users/<%= @user.id %>/posts/<%= @post.id %>/comments/new" class="border-4 border-solid border-black"> New comment</a>
+    </div>
   </div>
```
Comment (Munsa1) [inline app/views/posts/show.html.erb]: - [ ] Good job on making some changes however we  now have a different error in the post controller.

Screenshot
image

Videocast
[cinnamon-2023-12-23T235341+0200.webm](https://github.com/rmiyoyo/blog-app/assets/81932478/77d991e9-ea50-43a5-9a6b-3f5e1c0bcf21)

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @rmiyoyo, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
Passing Linters :heavy_check_mark: 
Correct GitHub Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/show.html.erb (hunk 2 above)
Comment (rmiyoyo) [reply app/views/posts/show.html.erb]: @H

> * [ ]  Good job on making some changes however we  now have a different error in the post controller.
> 
> Screenshot image
> 
> Videocast [cinnamon-2023-12-23T235341+0200.webm](https://github.com/rmiyoyo/blog-app/assets/81932478/77d991e9-ea50-43a5-9a6b-3f5e1c0bcf21)

Hi Munsa, I am out of reviews, do you have an idea how to solve the problem? I am currently on extra reviews

Comment (Munsa1) [reply app/views/posts/show.html.erb]: Hey @rmiyoyo, I wish I could help but I see the time, slack me if you have time

Comment (rmiyoyo) [reply app/views/posts/show.html.erb]: @Munsa1 I sent you a slack

**Comment by rmiyoyo** [comment]
# Work Done
- Fixed errors

N/B
I will work on improving the app after this review

**Review by julie-ify — APPROVED** [review]
Hi @rmiyoyo, 

## Highlights!
- Good code organization and working forms ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**